tests/test_encoding.rs

```diff
@@ -0,0 +1,362 @@
+
+        let proposals = vec![add, remove, update];
+        let (commit, welcome_option, _key_package_bundle_option) = group_state
+            .create_commit(&[], credential_bundle, proposals, true)
+            .unwrap();
+        let commit_encoded = commit.encode_detached().unwrap();
+        let commit_decoded = match MLSPlaintext::decode(&mut Cursor::new(&commit_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding MPLSPlaintext Commit: {:?}", err),
+        };
+
+        assert_eq!(commit, commit_decoded);
+
+        // Welcome messages
+
+        let welcome = welcome_option.unwrap();
+
+        let welcome_encoded = welcome.encode_detached().unwrap();
+        let welcome_decoded = match Welcome::decode(&mut Cursor::new(&welcome_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding Welcome message: {:?}", err),
+        };
+
+        assert_eq!(welcome, welcome_decoded);
+    }
+}
+
+#[test]
+fn test_welcome_message_encoding() {
+    let test_setup = create_encoding_test_setup();
+    let test_clients = test_setup.clients.borrow();
+    let alice = test_clients.get("alice").unwrap().borrow();
+
+    for group_state in alice.group_states.borrow_mut().values_mut() {
+        let credential_bundle = alice
+            .credential_bundles
+            .get(&group_state.ciphersuite().name())
+            .unwrap();
+
+        let capabilities_extension = Box::new(CapabilitiesExtension::new(
+            None,
+            Some(&[group_state.ciphersuite().name()]),
+            None,
+        ));
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        let key_package_bundle = KeyPackageBundle::new(
+            &[group_state.ciphersuite().name()],
+            credential_bundle,
+            mandatory_extensions,
+        )
+        .unwrap();
+
+        // Create a few proposals to put into the commit
+        let add = group_state.create_add_proposal(
+            &[],
+            credential_bundle,
+            key_package_bundle.get_key_package().clone(),
+        );
+
+        let proposals = vec![add];
+        let (_commit, welcome_option, _key_package_bundle_option) = group_state
+            .create_commit(&[], credential_bundle, proposals, true)
+            .unwrap();
+
+        // Welcome messages
+
+        let welcome = welcome_option.unwrap();
+
+        let welcome_encoded = welcome.encode_detached().unwrap();
+        let welcome_decoded = match Welcome::decode(&mut Cursor::new(&welcome_encoded)) {
+            Ok(a) => a,
+            Err(err) => panic!("Error decoding Welcome message: {:?}", err),
+        };
+
+        assert_eq!(welcome, welcome_decoded);
+    }
+}
+
+// TODO: Leaving out Extensions completely still yields a decoding
+// error! Should be solved by #164.
```

Comment:
This is probably also because the lifetime extension is invalid (see comment above).